toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,339 @@
+	// NetworkBootProtoStatic indicates that this is a static protocol
+	NetworkBootProtoStatic NetworkBootProto = "static"
+	// NetworkBootProtoBOOTP indicates that this is a bootp protocol
+	NetworkBootProtoBOOTP NetworkBootProto = "bootp"
+	// NetworkBootProtoiBFT indicates that this is a ibft protocol
+	NetworkBootProtoiBFT NetworkBootProto = "ibft"
+	// NetworkBootProtoDefault indicates that this is a DHCP protocol
+	NetworkBootProtoDefault NetworkBootProto = ""
+)
+
+// UnmarshalJSON Unmarshals a Network entry
+func (n *Network) UnmarshalJSON(b []byte) (err error) {
+	// Use an intermediate type which will use the default JSON unmarshal implementation
+	type IntermediateTypeNetwork Network
+	err = json.Unmarshal(b, (*IntermediateTypeNetwork)(n))
+	if err != nil {
+		return fmt.Errorf("failed to parse [Network]: %w", err)
+	}
+
+	// Now validate the resulting unmarshaled object
+	err = n.IsValid()
+	if err != nil {
+		return fmt.Errorf("failed to parse [Network]: %w", err)
+	}
+	return
+}
+
+// IsValid returns an error if the Network struct is not valid
+func (n *Network) IsValid() (err error) {
+	err = n.bootProtoIsValid()
+	if err != nil {
+		return
+	}
+
+	err = n.ipAddrIsValid()
+	if err != nil {
+		return
+	}
+
+	err = n.deviceIsValid()
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+// ConfigureNetwork performs network configuration during the kickstart unattended installation process
+func ConfigureNetwork(installChroot *safechroot.Chroot, systemConfig SystemConfig) (err error) {
+	const squashErrors = false
+	var dnsUpdate bool
+
+	for _, networkData := range systemConfig.Networks {
+		deviceName, err := checkNetworkDeviceAvailability(networkData)
+		if err != nil {
+			return err
+		} else if deviceName == "" {
+			err = fmt.Errorf("The input network device is currently not supported on this system")
+			return err
+		}
+
+		err = createNetworkConfigFile(installChroot, networkData, deviceName)
+		if err != nil {
+			return err
+		}
+
+		dnsUpdate = len(networkData.NameServers) > 0
+	}
+
+	err = shell.ExecuteLive(squashErrors, "systemctl", "restart", "systemd-networkd")
+	if err != nil {
+		logger.Log.Errorf("Unable to restart systemd-networkd: %s", err)
+		return
+	}
+
+	if dnsUpdate {
+		err = shell.ExecuteLive(squashErrors, "systemctl", "restart", "systemd-resolved")
+		if err != nil {
+			logger.Log.Errorf("Unable to restart systemd-resolved: %s", err)
+		}
+	}
+
+	return
+}
+
+func (n NetworkBootProto) string() string {
+	return fmt.Sprint(string(n))
+}
+
+func (n *Network) getValidNetworkBootProtos() (types []NetworkBootProto) {
+	return []NetworkBootProto{
+		NetworkBootProtoDHCP,
+		NetworkBootProtoStatic,
+		NetworkBootProtoBOOTP,
+		NetworkBootProtoiBFT,
+		NetworkBootProtoDefault,
+	}
+}
+
+// bootProtoIsValid returns an error if input bootproto is invalid
+func (n *Network) bootProtoIsValid() (err error) {
+	n.BootProto = strings.TrimSpace(n.BootProto)
+	for _, valid := range n.getValidNetworkBootProtos() {
```

Comment:
Instead of creating the list every time we get here, you could just add a global with it or even a global string -> boolean map, to make the code checking if the value is correct a one-liner like `if validNetworkBootProtos[n.BootProto]...`.

Reply:
Fixed